src/lib/PPrint.hs

```diff
@@ -720,18 +740,23 @@ instance Pretty (UDecl n l) where
     "effect" <+> p effName <> hardline <> foldMap (<>hardline) ops
     where ops = [ p pol <+> p (UAnnBinder b (unsafeCoerceE ty))
                  | (b, UEffectOpType pol ty) <- zip (toList $ fromNest opNames) opTys]
-  pretty (UHandlerDecl effName tyArgs _retEff retTy opDefs name) =
+  pretty (UHandlerDecl effName tyArgs retEff retTy opDefs name) =
     "handler" <+> p name <+> "of" <+> p effName <+> prettyBinderNest tyArgs
-    <+> ":" <+> "{todo: pretty effects}" <+> p retTy <> hardline
-    <> foldMap (<>hardline) ops
-    where ops = [ p rp <+> p n <+> "=" <+> p body
-                 | UEffectOpDef n rp body <- opDefs ]
+    <+> ":" <+> p retEff <+> p retTy <> hardline
+    <> foldMap ((<>hardline) . p) opDefs
+
+instance Pretty (UEffectOpDef n) where
+  pretty (UEffectOpDef rp n body) = p rp <+> p n <+> "=" <+> p body
+  pretty (UReturnOpDef body) = "return =" <+> p body
 
 instance Pretty UResumePolicy where
   pretty UNoResume = "jmp"
   pretty ULinearResume = "def"
   pretty UAnyResume = "ctl"
-  pretty UReturn = ""
+
+instance Pretty (UEffectRow n) where
+  pretty (EffectRow x Nothing) = encloseSep "<" ">" "," $ (p <$> toList x)
+  pretty (EffectRow x (Just y)) = "<" <> (hsep $ punctuate "," (p <$> toList x)) <+> "|" <+> p y <> ">"
```

Comment:
Dex uses curly braces for effect rows, so it would be better to keep it consistent in the pretty printer. We are planning to make a change to angle brackets, but it should happen both in the parser and in here at the same time.